meteor/sofie-server/data-access/repositories/mongo/mongo-segment-repository.ts

```diff
@@ -33,4 +35,22 @@ export class MongoSegmentRepository extends BaseMongoRepository implements Segme
 			})
 		)
 	}
+
+	public async deleteSegmentsForRundown(rundownId: string): Promise<void> {
+		this.assertDatabaseConnection(this.deleteSegmentsForRundown.name)
+		const segments: Segment[] = await this.getSegments(rundownId)
+
+		segments.forEach(async (segment: Segment) => this.partRepository.deletePartsForSegment(segment.id))
```

Comment:
You can avoid the "Promise returned from forEach argument is ignored" by employing: